attach/recvattach.c

```diff
@@ -553,28 +553,25 @@ static void query_pipe_attachment(const char *command, FILE *fp, struct Body *bo
       msgwin_clear_text();
       return;
     }
-    mutt_mktemp(tfile, sizeof(tfile));
-  }
-  else
-  {
-    tfile[0] = '\0';
+    buf_mktemp(tfile);
   }
 
-  if (mutt_pipe_attachment(fp, body, command, tfile))
+  if (mutt_pipe_attachment(fp, body, command, buf_strdup(tfile)))
```

Comment:
We want `buf_string()` here, or we'll leak

Reply:
Is it ok if I change `mutt_pipe_attachment()`?

```c
-int mutt_pipe_attachment(FILE *fp, struct Body *b, const char *path, char *outfile);
+int mutt_pipe_attachment(FILE *fp, struct Body *b, const char *path, const char *outfile);
```

Reply:
Yep, that's good

Reply:
fixed now